ddtrace/internal/datadog/profiling/dd_wrapper/test/test_forking.cpp

```diff
@@ -34,24 +38,68 @@ is_exit_normal(int status)
     return WIFEXITED(status) && WEXITSTATUS(status) == 0;
 }
 
+struct EmulateSamplerArg
+{
+    unsigned int id;
+    unsigned int sleep_time_ns;
+    std::atomic<bool>* done;
+};
+
+void*
+emulate_sampler_wrapper(void* argp)
+{
+    EmulateSamplerArg* arg = reinterpret_cast<EmulateSamplerArg*>(argp);
+
+    emulate_sampler(arg->id, arg->sleep_time_ns, *arg->done);
+
+    return nullptr;
+}
+
+void
+launch_pthread_samplers(const std::vector<EmulateSamplerArg>& args, std::vector<pthread_t>& threads)
+{
+    for (unsigned int i = 0; i < args.size(); i++) {
+        auto arg = args[i];
+        auto pthread_handle = threads[i];
+        pthread_create(&pthread_handle, nullptr, emulate_sampler_wrapper, reinterpret_cast<void*>(&arg));
```

Comment:
Should this be `&threads[i]` instead of `&pthread_handle`? IIUC `pthread_handle` is a local variable

Reply:
Oh, good catch.  If we want to use an intermediate variable for clarity, might have to do something like
```
auto *pthread_handle = &threads[i]
pthread_create(pthread_handle, ...
```

Reply:
oh, good catch. You're right, but I wonder how the test passes with this...

Reply:
Updated the code to use address of vec element